File: grind75/Dec20/235.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 
'TreeNode') -> 'TreeNode':
        if p.val > q.val:
            p, q = q, p

        if root.val < p.val:
            return self.lowestCommonAncestor(root.right, p, q)
        
        if root.val > q.val:
            return self.lowestCommonAncestor(root.left, p, q)
        
        return root

File: grind75/Dec20/235.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val   int
 *     Left  *TreeNode
 *     Right *TreeNode
 * }
 */

func lowestCommonAncestor(root, p, q *TreeNode) *TreeNode {
	val, lval, rval := root.Val, p.Val, q.Val
    if val > lval && val > rval {
        return lowestCommonAncestor(root.Left, p, q)
    }
    if val < lval && val < rval {
        return lowestCommonAncestor(root.Right, p, q)
    }
    return root

}
